### DISCUSSION TITLE
Neutral checks bypass Required status checks

### DISCUSSION THREAD

**Discussion by jonathanmorley** [community/community#45229] [Pull Requests]
### Select Topic Area

Bug

### Body

Setup:
1. Have a Status Check on a repository, which can report back one of 3 states: passing, failing, or neutral. Lets call this: `Check-1`
2. Have a Branch Protection Rule for a branch (e.g. `main`), which has a Required Status Check requiring `Check-1` to pass.

**Bug - Neutral checks count as 'passing'**

If you create a PR against `main`, crafted in such a way to trigger `Check-1` to return neutral, then the PR is mergeable.

This is not what the UI message says:
> Choose which [status checks](https://docs.github.com/rest/commits/statuses) must pass before branches can be merged into a branch that matches this rule

It is also not what the [Branch Protection Rule documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule) says:
> You can create a branch protection rule to enforce certain workflows for one or more branches, such as requiring an approving review or passing status checks for all pull requests merged into the protected branch.
[Troubleshooting required status checks](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks) does not mention Neutral statuses

There is another [discussion](https://github.com/orgs/community/discussions/24754) asking about this too, but the accepted answer is incorrect (it is quoting from the incorrect docs). That discussion contains a screenshot at the bottom, demonstrating the behaviour.

**Suggestion for a fix**

I do not recommend changing the behaviour, since that would be a breaking change to user's workflow. Instead, I suggest elaborating in the docs about the implications of a check returning a neutral status (and updating the other discussion to make it clear that the current 'Answer' is not correct).

**Comment by samertm** [comment]
This just bit me, would be nice if this was documented somewhere.
